libs/SalesforceSDKCore/SalesforceSDKCore/Classes/IDP/SFSDKIDPAuthClient.m

```diff
@@ -92,17 +96,24 @@ - (BOOL)refreshCredentials:(void (^) (SFOAuthInfo *, SFUserAccount *))successCal
 }
 
 - (BOOL)refreshCredentials {
-    
-    if (self.credentials.accessToken==nil) {
-        UIViewController<SFSDKLoginFlowSelectionView> *controller  = self.idpLoginFlowSelectionBlock();
-        controller.selectionFlowDelegate = self;
-        self.authWindow.viewController = controller;
-        [self.authWindow enable:YES withCompletion:nil];
+    BOOL result = NO;
+    if (self.credentials.accessToken==nil && self.config.idpEnabled) {
+        if ([self.config.idpDelegate respondsToSelector:@selector(authClientDisplayIDPLoginFlowSelection:)]) {
+           [self.config.idpDelegate authClientDisplayIDPLoginFlowSelection:self];
```

Comment:
If authClientDisplayIDPLoginFlowSelection is not defined on the delegate, nothing happens?